Add stats only and systematics settings option in statistical analysis

AvencastF opened the pull request:
Added `stats_only` argument in statistical analysis class and `compute_mu` function.

According to `iminuit` documentation, for statistical only fitting, systematics parameters can be fixed [[link](https://scikit-hep.org/iminuit/notebooks/basic.html#Fixing-and-releasing-parameters)]

jvdudley commented:
I think using `result.fixed` like [this](https://github.com/FAIR-Universe/HEP-Challenge/pull/74/files#diff-abed938b1a16a7f1ec0944c33eb6976f1132a9fe1519b298348dfa9f83d2dc8fR150-R155) is great, but instead of conditioning all of them on `stat_only`, I think it would be better to condition each one individually on the `TEST_SETTINGS[“systematics”]` in the [Starting Kit notebook](https://github.com/FAIR-Universe/HEP-Challenge/blob/d270acda503e5d6b471254c5fecdfe0e57f0174a/StartingKit_HiggsML_Uncertainty_Challenge.ipynb#L51-L57). I think that if participants have the ability to enable/disable individual systematics in the Starting Kit, it would be best for that to be reflected in the fitting.

AvencastF commented:
> I think using `result.fixed` like [this](https://github.com/FAIR-Universe/HEP-Challenge/pull/74/files#diff-abed938b1a16a7f1ec0944c33eb6976f1132a9fe1519b298348dfa9f83d2dc8fR150-R155) is great, but instead of conditioning all of them on `stat_only`, I think it would be better to condition each one individually on the `TEST_SETTINGS[“systematics”]` in the [Starting Kit notebook](https://github.com/FAIR-Universe/HEP-Challenge/blob/d270acda503e5d6b471254c5fecdfe0e57f0174a/StartingKit_HiggsML_Uncertainty_Challenge.ipynb#L51-L57). I think that if participants have the ability to enable/disable individual systematics in the Starting Kit, it would be best for that to be reflected in the fitting.

@jvdudley  Good idea. I've added corresponding functionalities. Please check.

Ragansu commented on sample_code_submission/model.py:
```diff
@@ -226,20 +233,21 @@ def balance_set(self):
         return balanced_set
 
-    def predict(self, test_set):
+    def predict(self, test_set, stat_only: bool = None, syst_settings: dict[str, bool] = None):
```
Are you sure this work without changing ingestion ? 
If not I would recommend using a different approach else it will be difficult to test it with codabench.org

AvencastF replied:
Yes. This works perfectly with the current ingestion program.

Ragansu commented on ingestion_program/ingestion.py:
```diff
@@ -114,19 +114,23 @@ def init_submission(self, Model):
         self.data.delete_train_set()
 
-    def fit_submission(self):
+    def fit_submission(self, **kwargs):
```
please don't change ingestion

Ragansu commented on ingestion_program/ingestion.py:
```diff
@@ -114,19 +114,23 @@ def init_submission(self, Model):
         """
         print("[*] Calling fit method of submitted model")
-        self.model.fit()
+        self.model.fit(**kwargs)
```
please don't change ingestion

AvencastF replied:
reverted

Ragansu commented:
Please merge the master to this branch and test it to be sure we changed quite a lot of things in master recently. 

Thanks

RC

Ragansu commented:
So for future reference. You might want to rebase so that your commits come above the commits from the master 
I used to make this same mistake. 

Since I did check it before I am going to merge it. 

RC

AvencastF commented:
@Ragansu I've tested the new dataset, and it can run smoothly.

Ragansu commented:
Perfect
